tests/query_workflow.go

```diff
@@ -146,26 +145,72 @@ func (s *ClientFunctionalSuite) TestQueryWorkflow_Consistent_PiggybackQuery() {
 }
 
 func (s *ClientFunctionalSuite) TestQueryWorkflow_QueryWhileBackoff() {
-	testname := s.T().Name()
-	workflowFn := func(ctx workflow.Context) (string, error) {
-		workflow.SetQueryHandler(ctx, testname, func() (string, error) {
+
+	testnameFail := "backoff_query_fail"
+	testnamePass := "backoff_query_pass"
+
+	workflowFnFail := func(ctx workflow.Context) (string, error) {
+		_ = workflow.SetQueryHandler(ctx, testnameFail, func() (string, error) {
 			return "should-reach-here", nil
 		})
-		return "", temporal.NewApplicationError("retry-me", "test-error")
+		return "", nil
+	}
+	workflowFnPass := func(ctx workflow.Context) (string, error) {
+		_ = workflow.SetQueryHandler(ctx, testnamePass, func() (string, error) {
+			return "should-reach-here", nil
+		})
+		return "", nil
 	}
 
+	testCases := []struct {
+		contextTimeout time.Duration
+		startDelay     time.Duration
+		expectError    bool
+		workflowFn     any
+		testname       string
+	}{
+		{
+			contextTimeout: time.Duration(10) * time.Second,
+			startDelay:     time.Duration(5) * time.Second,
+			expectError:    false,
+			workflowFn:     workflowFnFail,
+			testname:       testnameFail,
+		},
+		{
+			contextTimeout: time.Duration(8) * time.Second,
+			startDelay:     time.Duration(10) * time.Second,
+			expectError:    true,
+			workflowFn:     workflowFnPass,
+			testname:       testnamePass,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.testQueryWorkflow_QueryWhileBackoff(
+			tc.contextTimeout, tc.startDelay, tc.expectError, tc.workflowFn, tc.testname,
+		)
+	}
+}
+
+func (s *ClientFunctionalSuite) testQueryWorkflow_QueryWhileBackoff(
```

Comment:
Please inline this function in the test.

Reply:
done in next rev

Reply:
actually not happening...
doing this trigger linter error (defer inside the loop).
I will revert this change.